code/modules/mob/living/simple_animal/bot/construction.dm

```diff
@@ -177,7 +177,7 @@
 				playsound(loc, W.usesound, 100, 1)
 				var/turf/T = get_turf(user)
 				user << "<span class='notice'>You start attaching the gun to the frame...</span>"
-				sleep(40)
+				do_after(user, 40*W.toolspeed, 0, src, 1)
```

Comment:
Seemed to work as it was, I assume this makes 100% sure the do_after completes before progressing?